src/ILLink.RoslynAnalyzer/DynamicallyAccessedMembers/DynamicallyAccessedMembersSymbol.cs

```diff
@@ -0,0 +1,82 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.CodeAnalysis;
+using System.Text;
+using ILLink.Shared;
+using Microsoft.CodeAnalysis;
+
+namespace ILLink.RoslynAnalyzer
+{
+	public class DynamicallyAccessedMembersSymbol : ValueWithDynamicallyAccessedMembers
+	{
+		public readonly ISymbol Source;
+		public readonly bool IsMethodReturn;
+
+		public DynamicallyAccessedMembersSymbol (IMethodSymbol method, bool isMethodReturn) => (Source, IsMethodReturn) = (method, isMethodReturn);
+
+		public DynamicallyAccessedMembersSymbol (IParameterSymbol parameter) => Source = parameter;
+
+		public DynamicallyAccessedMembersSymbol (IFieldSymbol field) => Source = field;
+
+		public DynamicallyAccessedMembersSymbol (INamedTypeSymbol type) => Source = type;
+
+		// This ctor isn't used for dataflow - it's really just a wrapper
+		// for annotations on type arguments/parameters which are type-checked
+		// by the analyzer (outside of the dataflow analysis).
+		public DynamicallyAccessedMembersSymbol (ITypeSymbol typeArgument) => Source = typeArgument;
+
+		public DynamicallyAccessedMembersSymbol (ITypeParameterSymbol typeParameter) => Source = typeParameter;
+
+		public override DynamicallyAccessedMemberTypes DynamicallyAccessedMemberTypes =>
+			IsMethodReturn
+				? ((IMethodSymbol) Source).GetDynamicallyAccessedMemberTypesOnReturnType ()
+				: Source.GetDynamicallyAccessedMemberTypes ();
+
+		protected override Type EqualityContract => typeof (DynamicallyAccessedMembersSymbol);
+
+		public virtual bool Equals (DynamicallyAccessedMembersSymbol other)
+		{
+			return this == other || (other != null && EqualityContract == other.EqualityContract &&
+				EqualityComparer<ISymbol>.Default.Equals (Source, other.Source) &&
+				EqualityComparer<bool>.Default.Equals (IsMethodReturn, other.IsMethodReturn));
+		}
+
+		public override int GetHashCode ()
+		{
+			return HashUtils.Combine (EqualityContract, SymbolEqualityComparer.Default.GetHashCode (Source), IsMethodReturn);
+		}
+
+#if DEBUG
```

Comment:
Why debug only?
I mean I understand this is for debugging purposes, but I don't see almost any value in leaving it out of Release builds.